src/os/bluestore/BlueStore.cc

```diff
@@ -7237,6 +7237,10 @@ void BlueStore::_txc_finish_io(TransContext *txc)
     _txc_state_proc(&*p++);
   } while (p != osr->q.end() &&
 	   p->state == TransContext::STATE_IO_DONE);
+
+  if (osr->kv_submitted_waiters) {
```

Comment:
we can probably set the flag in the loop above when p->state >= STATE_KV_SUBMITTED and notify depending on this flag's value. just to avoid excessive wakeups.